### PR TITLE
feat(devices): add device deactivation via cira

### PR DESCRIPTION
**Pull request by bwendlandt-intel**
## PR Checklist

- [x] Unit Tests have been added for new changes
- [x] API tests have been updated if applicable
- [x] All commented code has been removed
- [x] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency.

## What are you changing?

- Add button to devices page to deactivate device
- Replace button on devices detail page from 'generate rpc deactivate cmd' to just deactivate device instead

### CODE REVIEW

**Review by tim-shockley — APPROVED** [review]

File: src/app/devices/device-toolbar/device-toolbar.component.ts [hunk 1]
```diff
@@ -104,17 +104,33 @@ export class DeviceToolbarComponent implements OnInit {
...
+            this.snackBar.open($localize`Error sending deactivation`, undefined, SnackbarDefaults.defaultError)
+            return of(null)
+          }),
+          finalize(() => {
+            this.isLoading = false
+          })
+        ).subscribe(data => {
+          void this.navigateTo('devices')
+          this.snackBar.open($localize`Deactivation sent successfully`, undefined, SnackbarDefaults.defaultSuccess)
+        }, err => {
```
[the remark this answers is not in the document]
Comment (bwendlandt-intel) [reply src/app/devices/device-toolbar/device-toolbar.component.ts:131]: Wouldn't the catch in pipe only catch the deactivation api call errors? This is after getting a successful deactivation, and would catch an error then? Like during a page nav fail?

[the remark this answers is not in the document]
Comment (tim-shockley) [reply src/app/devices/device-toolbar/device-toolbar.component.ts:131]: the catch in pipe will catch the same errors that the 'error' block will handle and in this case, the error is caught in the pipe clause, snackbar is shown, and the return statement 'of(2)' flows into the 'data' block of the subscribe, which shows a success message. I used stackblitz to do some scenarios and I'd suggest the following
`  this.devicesService
    .sendDeactivate(this.deviceId)
    .pipe(
      finalize(() => {
        this.isLoading = false
      }))
    .subscribe({
      next: (() => {
        this.snackBar.open($localize`Deactivation sent successfully`, undefined, SnackbarDefaults.defaultSuccess)
        void this.navigateTo('devices')
      }),
      error: ((err) => {
        console.log(err)
        this.snackBar.open($localize`Error sending deactivation`, undefined, SnackbarDefaults.defaultError)
      })
    })
`
[stackblitz](https://stackblitz.com/edit/typescript-f29srf?devToolsHeight=33&file=index.ts)

[the remark this answers is not in the document]
Comment (bwendlandt-intel) [reply src/app/devices/device-toolbar/device-toolbar.component.ts]: Super appreciate the examples, makes a lot of sense.  I'm still learning RXjs here and there.  Don't get a lot of time to get used to it.

I made your change and also applied it to the other similar deactivate call too.

**Review by tim-shockley — APPROVED** [review]
Looks good

**Pull request merged**